src/Cli/dotnet/SlnFileExtensions.cs

```diff
@@ -83,6 +83,27 @@ public static void AddProject(this SlnFile slnFile, string fullProjectPath, ILis
 
                 if (solutionFolders != null)
                 {
+                    if (solutionFolders.Any())
+                    {
+                        // Check for any existing projects if there is a new matching solution folder and add them to their own solution folder
+                        var duplicateProjects = slnFile.Projects.Where(p => solutionFolders.Contains(p.Name)
+                                                                        && p.TypeGuid != ProjectTypeGuids.SolutionFolderGuid).ToList();
+                        foreach (SlnProject duplicateProject in duplicateProjects)
+                        {
+                            slnFile.AddSolutionFolders(duplicateProject, new List<string>() { Path.GetDirectoryName(duplicateProject.FilePath) });
+                        }
+                    }
+                    else
+                    {
+                        // If the project conflicts with a solution folder, add it's own folder as a solution folder
```

Comment:
> add it's own folder as a solution folder

What does this mean? I am confused by this comment
```suggestion
                        // If the project has the same name of a solution folder, give it a solution folder named ??? to separate the two 
```